Add OpenAI plugin

ItsDysis opened the pull request:
I've created an OpenAI Plugin which makes the user able to use GPT-3 inside the Flowlauncher: https://github.com/ItsDysis/OpenAIFlowLauncher
Note: The user has to provide their own API key to use the plugin

jjw24 commented on plugins.json:
```diff
@@ -1196,5 +1196,17 @@
+        "Version": "1.0.0",
+        "Language": "python",
+        "Website": "https://github.com/ItsDysis/OpenAIFlowLauncher",
+        "UrlDownload": "https://github.com/ItsDysis/OpenAIFlowLauncher/archive/refs/tags/v1.0.0.zip"
+        "UrlSourceCode": "https://github.com/ItsDysis/OpenAIFlowLauncher"
```
```suggestion
        "UrlDownload": "https://github.com/ItsDysis/OpenAIFlowLauncher/archive/refs/tags/v1.0.0.zip",
        "UrlSourceCode": "https://github.com/ItsDysis/OpenAIFlowLauncher",
```

jjw24 commented:
The zip from the download url does not contain the 'lib' folder. I suspect you need to update this line here https://github.com/ItsDysis/OpenAIFlowLauncher/blob/0fb098e1b03da0eacd79a4a45218a4786f63704c/.github/workflows/publish_release.yml#L38. Once done you can do another release and update the json here.

ItsDysis commented:
> The zip from the download url does not contain the 'lib' folder. I suspect you need to update this line here https://github.com/ItsDysis/OpenAIFlowLauncher/blob/0fb098e1b03da0eacd79a4a45218a4786f63704c/.github/workflows/publish_release.yml#L38. Once done you can do another release and update the json here.

You were right, I've changed it and updated my branch.

jjw24 commented:
@ItsDysis you will need to move this [line]( https://github.com/ItsDysis/OpenAIFlowLauncher/blob/f1978bca32bdf17879d364233222444e125c97ff/main.py#L5) below the path setup at [here](https://github.com/ItsDysis/OpenAIFlowLauncher/blob/f1978bca32bdf17879d364233222444e125c97ff/main.py#L11) so everything in plugin.OpenAI can be imported from the lib folder.

After doing it I still run into module import issue below. I recommend copying your plugin to Flow's plugin folder and test it that way. You can also edit the plugin there while flow is running.

`Flow.Launcher.Core.ExternalPlugins.FlowPluginException
  HResult=0x80131500
  Message=Traceback (most recent call last):
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\numpy\core\__init__.py", line 23, in <module>
    from . import multiarray
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\numpy\core\multiarray.py", line 10, in <module>
    from . import overrides
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\numpy\core\overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\main.py", line 12, in <module>
    from plugin.OpenAI import OpenAI
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\plugin\OpenAI.py", line 4, in <module>
    import openai
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\openai\__init__.py", line 8, in <module>
    from openai.api_resources import (
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\openai\api_resources\__init__.py", line 7, in <module>
    from openai.api_resources.embedding import Embedding  # noqa: F401
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\openai\api_resources\embedding.py", line 4, in <module>
    import numpy as np
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\numpy\__init__.py", line 140, in <module>
    from . import core
  File "C:\Workbench\Git\Flow.Launcher\Output\Debug\UserData\Plugins\Flow.Launcher.Plugin.OpenAI\lib\numpy\core\__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError: `

Garulf commented:
Numpy is not a pure python module. May run into issues if the user is using a different python version.

jjw24 commented on plugins.json:
```diff
@@ -1208,7 +1208,18 @@
         "IcoPath": "https://cdn.jsdelivr.net/gh/Jessuhh/discord-timestamps-flowlauncher-plugin/app.png",
-        "DateAdded": "2022-12-05T21:48:10Z",
-        "LatestReleaseDate": "2022-12-03T21:29:20Z"
+        "DateAdded": "2022-12-05T21:48:10Z"
```
```suggestion
        "DateAdded": "2022-12-05T21:48:10Z",
        "LatestReleaseDate": "2022-12-03T21:29:20Z"
```

Garulf commented:
> Numpy is not a pure python module. May run into issues if the user is using a different python version.

https://github.com/ItsDysis/OpenAIFlowLauncher/blob/f1978bca32bdf17879d364233222444e125c97ff/.github/workflows/publish_release.yml#L14
Please change this line to Python version `3.8`. It wont make numpy work across versions but will build on Flow's preferred version. 

Alternatively you can look into packaging your plugin using a program like: [Nuitka](https://github.com/Nuitka/Nuitka), or [PyInstaller](https://www.pyinstaller.org/). These programs allow you to package your own python with your plugin allowing you to not worry about what Python version a user may have configured.

Garulf commented:
Closing due to no activity. Please reopen if needed.
